solution

MaksymMatuliak opened the pull request.

MaksymMatuliak committed:
solution

Sofasmile commented on src/main/java/homework/Main.java:
```diff
@@ -0,0 +1,19 @@
+        FirstThread firstThread = new FirstThread(numbers);
+        SecondThread secondThread = new SecondThread(numbers);
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+        executorService.invokeAll(List.of(firstThread, secondThread));
```
суть завдання полягає в тому, щоб сума елементів одного списку обчислювалась в кількох потоках паралельно, і на виході ми отримували одне число = firstThreadRes + secondThreadRes.... 
В даному випадку в тебе сума результатів обчислень в firstThread і secondThread рівні і в сумі не дають правильного результату

MaksymMatuliak replied:
Не впевнений що справді до кінця зрозумів...

MaksymMatuliak committed:
treads count only a half of list

Sofasmile commented on src/main/java/homework/FirstThread.java:
```diff
@@ -0,0 +1,21 @@
+    public Integer call() {
+        Integer sum = randomNumberList.getNumbers()
+                .stream()
+                .limit(randomNumberList.getNumbers().size() / 2 + 1)
```
дане рішення не зовсім універсальне, адже для того щоб здійснити обчислення в 5 потоках - потрібно буде створювати додаткові класи і по іншому реалізовувати роботу методу call
спробуй зробити так, щоб клас був один, але об'єктів класу могло бути багато

MaksymMatuliak committed:
added universal thread

MaksymMatuliak committed:
deleted magic numbers

Sofasmile commented on src/main/java/homework/Main.java:
```diff
@@ -0,0 +1,29 @@
+    public static void main(String[] args) throws InterruptedException, ExecutionException {
+        RandomNumberList numbers = new RandomNumberList();
+        ThreadCounter threadCounter =
+                new ThreadCounter(numbers, 1, 2);
```
magic numbers) let's create a constant

MaksymMatuliak replied:
It's weird I thought that I pushed it yesterday

MaksymMatuliak committed:
deleted magic

MaksymMatuliak commented:
I don't understand how, but commit from yesterday where I deleted magic numbers appeared before your commits (I couldn't push anything so that's why I deleted one '-' in the last commit" and it refreshed it I think)

Sofasmile commented on src/main/java/homework/Main.java:
```diff
@@ -8,19 +8,23 @@
 import java.util.concurrent.Future;
 
 public class Main {
+    private static final int numbersOfTreads = 2;
```
https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names

MaksymMatuliak replied:
done

Sofasmile commented on src/main/java/homework/Main.java:
```diff
@@ -8,19 +8,23 @@
 public class Main {
+    private static final int numbersOfTreads = 2;
+    private static final int firstPart = 1;
+    private static int secondPart = 2;
```
```suggestion
    private static final int secondPart = 2;
```

MaksymMatuliak replied:
done

MaksymMatuliak committed:
fixed style

MaksymMatuliak committed:
fixed bug in thread
